agent/consul/rate/handler.go

```diff
@@ -169,11 +169,15 @@ func (h *Handler) Run(ctx context.Context) {
 // Allow returns an error if the given operation is not allowed to proceed
 // because of an exhausted rate-limit.
 func (h *Handler) Allow(op Operation) error {
-	// TODO(NET-1383): actually implement the rate limiting logic.
-	//
-	// Example:
-	//	if !h.limiter.Allow(globalWrite) {
-	//	}
+	existingCfg := h.cfg.Load()
+	if existingCfg.GlobalMode == ModeDisabled {
+		return nil
+	}
```

Comment:
nit: What's the reason for the `existing` prefix here? I think `cfg` would probably be clear enough?